apps/minaintyg/src/pages/Certificates/components/CertificateList.tsx

```diff
@@ -0,0 +1,33 @@
+import { getYear, parseISO } from 'date-fns'
+import { CertificateListItem } from '../../../schema/certificateList.schema'
+import { CertificateCard } from './CertificateCard/CertificateCard'
+import { SortingOrder } from './CertificateListOrder'
+
+const getCertificateYear = ({ timestamp }: CertificateListItem) => getYear(parseISO(timestamp))
+
+export function CertificateList({ certificates, order = 'descending' }: { certificates: CertificateListItem[]; order?: SortingOrder }) {
+  const years = certificates.reduce((result, certificate) => result.add(getCertificateYear(certificate)), new Set<number>())
+  return (
+    <div>
+      {Array.from(years)
+        .sort((a, b) => (order === 'ascending' ? a - b : b - a))
+        .map((year) => (
+          <div key={year} className="[&:not(:last-child)]:mb-10" data-testid={`${year}-certificates`}>
```

Comment:
Can this be extracted into its own component to add readability?

Reply:
Rather than naming this div, I have moved the business logic out to a hook. The component should be much more readable now.